code/modules/research/server.dm

```diff
@@ -182,11 +111,11 @@
 		servers += list(list(
 			"name" = S.name,
 			"server_id" = S.server_id,
-			"temperature" = S.get_env_temp(),
-			"temperature_warning" = S.temp_tolerance_high,
-			"temperature_max" = S.temp_tolerance_damage,
-			"enabled" = !S.research_disabled,
-			"overheated" = S.overheated,
+			"temperature" = S.get_temperature(),
+			"temperature_warning" = S.get_warning_temperature(), // TODO: CHANGE IT TO A VARIABLE
+			"temperature_max" = S.get_overheat_temperature(),
+			"enabled" = !(S.machine_stat & TURNED_OFF),
```

Comment:
is this only limited to `TURNED_OFF`? I think other flags can block this enability.

Reply:
this is to display state of the power button as you can turn on/off servers.

Reply:
then please leave a commont here for that.

Reply:
done